README.rst
********************
Geolocation contexts
********************
Allows you to show pages and content elements for people in certain
countries, continents or within a given area.

It uses the `Maxmind GeoIP`__ database to determine the user's probable
physical location.
Note that this is working most of the time, but also fails regularly.

Country and Continent rules are pretty safe to use, while small distances
are not accurate at all.

__ http://www.maxmind.com/en/geolocation_landing


=====
Setup
=====
#. Install the PHP extension ``geoip`` (requires root access to the server)

   If that is not possible, install the ``Net_GeoIP`` package from PEAR
#. Install and activate this TYPO3 extension ``contexts_geolocation``


Dependencies
============
- ``contexts`` TYPO3 extension
- ``static_info_tables`` TYPO3 extension
- ``geoip`` PHP extension, see `its homepage`__
- or `Net_GeoIP`__ from PEAR instead of ``geoip``

__ http://pecl.php.net/package/geoip
__ http://pear.php.net/package/Net_GeoIP


=============
Context types
=============
The ``contexts_geolocation`` extension ships with three different
context types:

Continent
=========
Matches when the user is on one of the selected continents.

Continent ``- unknown -`` can be used to match all users whose position
could not be determined.

Country
=======
Matches when the user accessing the page is in one of the selected countries.

Country ``- unknown -`` can be used to match all users whose position
could not be determined.

Distance
========
Select a location on a map and a distance, and the context matches when
the user's location is within this area.

Note that the free geoip database does seldom give you exact locations,
so this is more a fun experiment rather that it should be used in production.

.. image:: doc/cfg-distance.png


=======
Plugins
=======

Contexts geolocation: show current position
===========================================
Shows a map with a marker at the detected position of the user's IP.

Also shows a dump of other data for this IP, like continent, country and city.

Useful for testing and debugging.

Allows manual IP address input.


ChangeLog
2015-06-22  Christian Weiske  <christian.weiske@netresearch.de>

	* Fix class names to for TYPO3 4.5 autoloader
	* Version 0.4.1

2014-03-07  Christian Weiske  <christian.weiske@netresearch.de>

	* Set minimum TYPO3 version to 4.5.0
	* Version 0.4.0

2013-11-22  Christian Weiske  <christian.weiske@netresearch.de>

	* Remove manual.sxw; it is not needed by TER anymore
	* Set minimum TYPO3 version to 4.6.0
	* Version 0.3.1

2013-08-16  Christian Weiske  <christian.weiske@netresearch.de>

	* Use Net_GeoIP from PEAR when available
	* Fix crash when geoip extension is not loaded
	* Add local leaflet.js
	* Add automatic map update when manually inserting lon/lat values
	* Add clickability to backend map
	* Version 0.3.0

2013-07-21  Rico Sonntag  <rico.sonntag@netresearch.de>

	* Fix fatal error in frontend plugin when geoip extension
	  is not available

2013-06-03  Christian Weiske  <christian.weiske@netresearch.de>

	* Add TYPO3 dependency to ext_emconf
	* Add Changelog
	* Fix icon for TYPO3 4.x
	* Version 0.2.1

2013-05-31  Christian Weiske  <christian.weiske@netresearch.de>

	* First public release
	* Version 0.2.0


ext_emconf.php
<?php

/***************************************************************
 * Extension Manager/Repository config file for ext "contexts_geolocation".
 *
 * Auto generated 07-03-2014 14:39
 *
 * Manual updates:
 * Only the data in the array - everything else is removed by next
 * writing. "version" and "dependencies" must not be touched!
 ***************************************************************/

$EM_CONF[$_EXTKEY] = array(
	'title' => 'Multi-channel contexts: Geolocation',
	'description' => 'Use the user\'s geographic location as context',
	'category' => 'misc',
	'author' => 'Christian Weiske, Marian Pollzien, Rico Sonntag',
	'author_email' => 'typo3.org@netresearch.de',
	'shy' => '',
	'dependencies' => 'contexts,static_info_tables',
	'conflicts' => '',
	'priority' => '',
	'module' => '',
	'state' => 'alpha',
	'internal' => '',
	'uploadfolder' => 0,
	'createDirs' => 'typo3temp/contexts_geolocation/',
	'modify_tables' => '',
	'clearCacheOnLoad' => 1,
	'lockType' => '',
	'author_company' => 'Netresearch GmbH & Co.KG',
	'version' => '0.4.1',
	'constraints' => array(
		'depends' => array(
			'typo3' => '4.5.0-6.2.99',
			'contexts' => '0.4.0-',
			'static_info_tables' => '2.0.0-',
		),
		'conflicts' => array(
		),
		'suggests' => array(
		),
	),
	'_md5_values_when_last_written' => 'a:32:{s:9:"ChangeLog";s:4:"1afe";s:16:"ext_autoload.php";s:4:"37ca";s:21:"ext_conf_template.txt";s:4:"ab52";s:16:"ext_contexts.php";s:4:"665f";s:12:"ext_icon.gif";s:4:"e433";s:17:"ext_localconf.php";s:4:"fef1";s:14:"ext_tables.php";s:4:"4aa3";s:11:"README.html";s:4:"e74c";s:10:"README.rst";s:4:"8419";s:12:"Settings.yml";s:4:"1ff9";s:19:"Classes/Adapter.php";s:4:"28ed";s:19:"Classes/Backend.php";s:4:"4795";s:21:"Classes/Exception.php";s:4:"fd4e";s:25:"Classes/Adapter/GeoIp.php";s:4:"4a63";s:28:"Classes/Adapter/NetGeoIp.php";s:4:"c4c7";s:34:"Classes/Context/Type/Continent.php";s:4:"d172";s:32:"Classes/Context/Type/Country.php";s:4:"9d2a";s:33:"Classes/Context/Type/Distance.php";s:4:"fa86";s:36:"Configuration/flexform/Continent.xml";s:4:"dcd7";s:34:"Configuration/flexform/Country.xml";s:4:"6c20";s:35:"Configuration/flexform/Distance.xml";s:4:"35af";s:39:"Resources/Private/Language/flexform.xml";s:4:"3d3b";s:43:"Resources/Private/Language/locallang_db.xml";s:4:"e31a";s:47:"Resources/Public/JavaScript/Leaflet/leaflet.css";s:4:"8ae0";s:50:"Resources/Public/JavaScript/Leaflet/leaflet.ie.css";s:4:"db9b";s:46:"Resources/Public/JavaScript/Leaflet/leaflet.js";s:4:"df34";s:53:"Resources/Public/JavaScript/Leaflet/images/layers.png";s:4:"2ba2";s:58:"Resources/Public/JavaScript/Leaflet/images/marker-icon.png";s:4:"87f6";s:61:"Resources/Public/JavaScript/Leaflet/images/marker-icon@2x.png";s:4:"1c82";s:60:"Resources/Public/JavaScript/Leaflet/images/marker-shadow.png";s:4:"e7bd";s:20:"doc/cfg-distance.png";s:4:"4945";s:44:"pi/class.tx_contextsgeolocation_position.php";s:4:"82b4";}',
	'suggests' => array(
	),
);

?>